pkg/catalog/catalog.go

```diff
@@ -2845,15 +2844,15 @@ func getHashSum(value, signingKey []byte) []byte {
 	return h.Sum(nil)
 }
 
-func (c *Catalog) VerifyLinkAddress(repository, branch, path, physicalAddress string) error {
+func (c *Catalog) VerifyLinkAddress(repo *Repository, branch, path, physicalAddress string) error {
 	idx := strings.LastIndex(physicalAddress, LinkAddressSigningDelimiter)
 	if idx < 0 {
 		return fmt.Errorf("address is not signed: %w", graveler.ErrLinkAddressInvalid)
 	}
 	address := physicalAddress[:idx]
 	signature := physicalAddress[idx+1:]
 
-	stringToVerify, err := getAddressJSON(repository, branch, path, address)
+	stringToVerify, err := getAddressJSON(repo.Name, branch, path, address)
```

Comment:
Again, not related to this PR - 
But having both ID and Name of a Repository meaning the same,
It's very confusing.

Reply:
Not part of this PR